kex.c

```diff
@@ -606,7 +606,7 @@ kex_input_kexinit(int type, u_int32_t seq, struct ssh *ssh)
 	}
 	ssh_dispatch_set(ssh, SSH2_MSG_KEXINIT, NULL);
 	ptr = sshpkt_ptr(ssh, &dlen);
```

Comment:
This can return a NULL pointer, if the buffer fails the sanity check.  It seems unlikely but is possible.